cli/commands/plan.go

```diff
@@ -238,8 +240,13 @@ func (cmd *planHandler) handleStart(a *kingpin.Application, e *kingpin.ParseElem
 func printStatus(planName string, rawJSON bool) {
 	client.SetCustomResponseCheck(checkPlansResponse)
 	responseBytes, err := client.HTTPServiceGet(fmt.Sprintf("v1/plans/%s", planName))
+
 	if err != nil {
-		client.PrintMessageAndExit(err.Error())
+		if err.Error() == "plan has errors" {
+			defer client.Exit(1)
```

Comment:
Why an exit code of 1?  Client didn't have a problem..

Also we whould set constants for exit codes if we're going to have them be meaningful.

Reply:
Looking at the code, we use `PrintMessageAndExit` if an error is returned (and always exit with an exit code of 1: https://github.com/mesosphere/dcos-commons/pull/1664/files#diff-fe7dfb33c57e6046f4d1d53c9af885c6R33).

We call this termination function for many error cases, some of which are not client specific. But you're right in that we probably shouldn't use it here. We have a success condition, and tooling can check the json output of the command to determine if the plan has errors and not have to rely on the (undefined) status code.

This does make me wonder why it was needed to change the plan endpoint to return a 4xx HTTP code, as the error part of the JSON object returned? Is there a case where returning a 202 (as was the case originally) is incorrect?